dali/operators/random/rng_base_gpu.cuh

```diff
@@ -64,9 +113,11 @@ __global__ void RNGKernel(BlockDesc<IsNoiseGen>* __restrict__ block_descs,
   int blk_stride = blockDim.y * gridDim.y;
   int blk = blockIdx.y * blockDim.y + threadIdx.y;
   for (; blk < nblocks; blk += blk_stride) {
-    auto desc = block_descs[blk];
-    Dist dist = DefaultDist ? Dist() : dists[desc.sample_idx];
-    Generate<T, Dist>(desc, dist, rng);
+    auto block = block_descs[blk];
+    auto sample = sample_descs[block.sample_idx];
```

Comment:
Divided into BlockDesc and SampleDesc, because we added some info that doesn't need to be repeated for every block (pixel stride, channel stride, number of channels...)